tools/speech_data_explorer/data_explorer.py

```diff
@@ -1,846 +1,1044 @@
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-import argparse
-import base64
-import csv
-import datetime
-import difflib
-import io
-import json
-import math
-import operator
-import os
-import pickle
-from collections import defaultdict
-from os.path import expanduser
-from pathlib import Path
-
-import dash
-import dash_bootstrap_components as dbc
-import diff_match_patch
-import editdistance
-import jiwer
-import librosa
-import numpy as np
-import soundfile as sf
-import tqdm
-from dash import dash_table, dcc, html
-from dash.dependencies import Input, Output, State
-from dash.exceptions import PreventUpdate
-from plotly import express as px
-from plotly import graph_objects as go
-from plotly.subplots import make_subplots
-
-# number of items in a table per page
-DATA_PAGE_SIZE = 10
-
-# operators for filtering items
-filter_operators = {
-    '>=': 'ge',
-    '<=': 'le',
-    '<': 'lt',
-    '>': 'gt',
-    '!=': 'ne',
-    '=': 'eq',
-    'contains ': 'contains',
-}
-
-# parse table filter queries
-def split_filter_part(filter_part):
-    for op in filter_operators:
-        if op in filter_part:
-            name_part, value_part = filter_part.split(op, 1)
-            name = name_part[name_part.find('{') + 1 : name_part.rfind('}')]
-            value_part = value_part.strip()
-            v0 = value_part[0]
-            if v0 == value_part[-1] and v0 in ("'", '"', '`'):
-                value = value_part[1:-1].replace('\\' + v0, v0)
-            else:
-                try:
-                    value = float(value_part)
-                except ValueError:
-                    value = value_part
-            return name, filter_operators[op], value
-    return [None] * 3
-
-
-# standard command-line arguments parser
-def parse_args():
-    parser = argparse.ArgumentParser(description='Speech Data Explorer')
-    parser.add_argument(
-        'manifest', help='path to JSON manifest file',
-    )
-    parser.add_argument('--vocab', help='optional vocabulary to highlight OOV words')
-    parser.add_argument('--port', default='8050', help='serving port for establishing connection')
-    parser.add_argument(
-        '--disable-caching-metrics', action='store_true', help='disable caching metrics for errors analysis'
-    )
-    parser.add_argument(
-        '--estimate-audio-metrics',
-        '-a',
-        action='store_true',
-        help='estimate frequency bandwidth and signal level of audio recordings',
-    )
-    parser.add_argument(
-        '--audio-base-path',
-        default=None,
-        type=str,
-        help='A base path for the relative paths in manifest. It defaults to manifest path.',
-    )
-    parser.add_argument('--debug', '-d', action='store_true', help='enable debug mode')
-    args = parser.parse_args()
-
-    # assume audio_filepath is relative to the directory where the manifest is stored
```

Comment:
Please integrate Ante's updates back (to support audio_base_path)

Reply:
done